ksql-metastore/src/main/java/io/confluent/ksql/metastore/MetaStoreImpl.java

```diff
@@ -83,8 +83,18 @@ public StructuredDataSource getSource(final String sourceName) {
   public Optional<StructuredDataSource> getSourceForTopic(final String ksqlTopicName) {
     return dataSources.values()
         .stream()
-        .filter(p -> p.source.getKsqlTopic().getName() != null
-            && p.source.getKsqlTopic().getName().equals(ksqlTopicName))
+        .filter(p -> p.source.getTopicName() != null
+            && p.source.getTopicName().equals(ksqlTopicName))
```

Comment:
From the code it's pretty unclear what the difference between `getTopicName` and `getKafkaTopicName` are. I know it's not your code and it's probably out of scope for this PR but I think it'd be good to get a better naming scheme down so that future bugs around this are avoided. Perhaps something like renaming `KsqlTopic` to `KsqlTopicAlias`, `KsqlTopicReference` or something like that to avoid overloading "name".

Reply:
I agree @agavra , if you remember our past discussion, we have KSQL topic which is defined for a Kafka topic. I think renaming `topicName` to `ksqlTopicName` and keeping `kafkaTopicName` would help reducing ambiguity here.

Reply:
Completely agree, but want to keep this diff small since it's going into a branch past code freeze. Will do in a follow-up.